common/src/chain/transaction/signature/inputsig/authorize_pubkey_spend.rs

```diff
@@ -72,200 +72,107 @@ mod test {
-                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap_err(),
-                TransactionSigError::InvalidSignatureEncoding
-            ));
-        }
-        {
-            let sighash_type =
-                SigHashType::try_from(SigHashType::ALL | SigHashType::ANYONECANPAY).unwrap();
-            let witness = prepare_data_for_wrong_destination(sighash_type);
-            assert!(matches!(
-                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap_err(),
-                TransactionSigError::InvalidSignatureEncoding
-            ));
-        }
-        {
-            let sighash_type = SigHashType::try_from(SigHashType::NONE).unwrap();
-            let witness = prepare_data_for_wrong_destination(sighash_type);
-            assert!(matches!(
-                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap_err(),
-                TransactionSigError::InvalidSignatureEncoding
-            ));
-        }
-        {
-            let sighash_type =
-                SigHashType::try_from(SigHashType::NONE | SigHashType::ANYONECANPAY).unwrap();
-            let witness = prepare_data_for_wrong_destination(sighash_type);
-            assert!(matches!(
-                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap_err(),
-                TransactionSigError::InvalidSignatureEncoding
-            ));
-        }
-        {
-            let sighash_type = SigHashType::try_from(SigHashType::SINGLE).unwrap();
-            let witness = prepare_data_for_wrong_destination(sighash_type);
-            assert!(matches!(
-                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap_err(),
-                TransactionSigError::InvalidSignatureEncoding
-            ));
-        }
-        {
-            let sighash_type =
-                SigHashType::try_from(SigHashType::SINGLE | SigHashType::ANYONECANPAY).unwrap();
-            let witness = prepare_data_for_wrong_destination(sighash_type);
-            assert!(matches!(
-                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap_err(),
-                TransactionSigError::InvalidSignatureEncoding
-            ));
+    fn invalid_signature() {
+        let (private_key, public_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+        let destination = Destination::PublicKey(public_key.clone());
+        let tx = generate_unsigned_tx(&destination, 1, 2).unwrap();
+
+        for sighash_type in sig_hash_types() {
+            let witness = StandardInputSignature::produce_signature_for_input(
+                &private_key,
+                sighash_type,
+                destination.clone(),
+                &tx,
+                INPUT_NUM,
+            )
+            .unwrap();
+            let mut raw_signature = witness.get_raw_signature().clone();
+            raw_signature[0] = raw_signature[0].wrapping_add(2);
```

Comment:
add a comment that this modifies the enum arm of ...

Reply:
As asked on slack, this isn't incomplete. Your wrapping add or add here changes the first byte in the serialized form of the signature, and hence it doesn't change the signature but the enum arm of the signature type. All I'm saying that you should write a comment explaining that because it's not so obvious. You can also write a function with a proper name to describe this operation.